### PR TITLE
Delete1

### PR DESCRIPTION
**Pull request by pmuir**
#### Submitter checklist

- [ ] Change is code complete and matches issue description.
- [ ] Change is covered by existing or new tests.

#### Description
implement `jx delete app`

#### Which issue this PR fixes

fixes #

### CODE REVIEW

File: pkg/jx/cmd/delete_application.go [hunk 1]
```diff
@@ -312,29 +312,28 @@ func (o *DeleteApplicationOptions) waitForGitOpsPullRequest(env *v1.Environment,
 			if pr.Merged != nil && *pr.Merged {
 				log.Infof("Request %s is merged!\n", util.ColorInfo(pr.URL))
 				return nil
+			}
+			if pr.IsClosed() {
```
Comment (jtnord) [inline pkg/jx/cmd/delete_application.go]: is this applications or apps?   I am a little confused as to what is this PR is trying to acheive - and exepected it from the first few files to be "deleting (aka uninstalling) apps"

File: pkg/jx/cmd/delete_application.go [hunk 2]
```diff
@@ -312,29 +312,28 @@ func (o *DeleteApplicationOptions) waitForGitOpsPullRequest(env *v1.Environment,
...
-							if err != nil {
-								if !logMergeFailure {
-									logMergeFailure = true
-									log.Warnf("Failed to merge the Pull Request %s due to %s maybe I don't have karma?\n", pr.URL, err)
-								}
+				if status == "success" {
+					if !o.NoMergePullRequest {
+						err = gitProvider.MergePullRequest(pr, "jx promote automatically merged promotion PR")
+						if err != nil {
+							if !logMergeFailure {
```
Comment (jtnord) [inline pkg/jx/cmd/delete_application.go]: it looks like this boolean is the inverse of what it is controlling (if it is false then the message is logged, if true then it is not logged.

**Comment by pmuir** [comment]
> Can you write a description in the {PR) (even more than a one word title would help). the code looks mostly ok, but I do not know what the intent of this PR is in order to say anything concrete

Oh, normally github copies the commit message to the PR - this implements `jx delete app`

File: pkg/jx/cmd/delete_application.go (hunk 1 above)
Comment (pmuir) [reply pkg/jx/cmd/delete_application.go]: Hmm, possibly something went wrong, I meant to do the unalias as a single commit. Let me look.

Comment (pmuir) [reply pkg/jx/cmd/delete_application.go]: Oh, this looks like a reformat of a file that got caught up. Let me remove it.

File: pkg/jx/cmd/delete_application.go (hunk 2 above)
Comment (pmuir) [reply pkg/jx/cmd/delete_application.go]: Was a reformat

**Pull request merged**